### PR TITLE
feat: gather AWS Lambda metadata on first function invocation

### PR DESCRIPTION
**Pull request by trentm**
This uses the new 'expectExtraMetadata' and '.setExtraMetadata()'
functionality on the http-client v10.3.0.

Fixes: #2404

(This depends on the http-client changes in https://github.com/elastic/apm-nodejs-http-client/pull/170.)

### Checklist

- [x] Implement code
- [x] Add tests
- [x] Add CHANGELOG.asciidoc entry
- [x] Commit message follows [commit guidelines](https://github.com/elastic/apm-agent-nodejs/blob/master/CONTRIBUTING.md#commit-message-guidelines)

### CODE REVIEW

File: lib/lambda.js [hunk 1]
```diff
@@ -96,6 +138,15 @@ function elasticApmAwsLambda (agent) {
 
     return function wrappedLambda (payload, context, callback) {
       log.trace({ awsRequestId: context && context.awsRequestId }, 'lambda: fn start')
+
+      if (isColdStart) {
```
Comment (astorm) [inline lib/lambda.js:142]: Just noting that https://github.com/elastic/apm-agent-nodejs/pull/2194 is also detecting a coldstart -- whomever gets in second will want to make sure their code still plays nice with the other PR.

File: package.json [hunk 2]
```diff
@@ -87,7 +87,7 @@
     "basic-auth": "^2.0.1",
     "cookie": "^0.4.0",
     "core-util-is": "^1.0.2",
-    "elastic-apm-http-client": "^10.2.0",
+    "elastic-apm-http-client": "elastic/apm-nodejs-http-client#trentm/lambda-delayed-metadata",
```
Comment (astorm) [inline package.json:90]: Just a note that we'll want to change this once the client PR lands.

**Review by astorm — APPROVED** [review]
Looks good -- approving (pending client code landing and not requiring additional changes) 👍

File: lib/lambda.js (hunk 1 above)
Comment (trentm) [reply lib/lambda.js:142]: Yup, noted. Let me know if you have a preference. I have the smaller diff here, so I'm happy to wait and handle the merge conflict, if you prefer.

**Comment by trentm** [comment]
I retested manually with the merged update.

**Review by astorm — COMMENTED** [review]
👍

**Pull request merged**